### DISCUSSION TITLE
How to pass dynamic args to UseContractWrite

### DISCUSSION THREAD

**Discussion by 0xTimepunk** [wevm/wagmi#948] [Q&A] [answered]
I have arguments that are validated in a form before call of a certain smart contract function.

Since these values are dynamic, write from useContractWrite is appearing as undefined.

What's the correct way to set these dynamic arguments?

**Comment by jxom** [comment, marked as answer]
I posted a code snippet here of how to use dynamic args with `usePrepareContractWrite`. This is definitely something that needs to be documented in the "Examples" section though.

https://github.com/wagmi-dev/wagmi/discussions/782#discussioncomment-3289133

**Comment by jxom** [comment]
There is now an official example for this!

https://wagmi.sh/examples/contract-write-dynamic

**Reply by 0xTimepunk** [reply]
Thanks @jxom
